internal/models/challenge.go

```diff
@@ -2,19 +2,23 @@ package models
 
 import (
 	"database/sql"
+	"github.com/go-webauthn/webauthn/protocol"
+	"github.com/go-webauthn/webauthn/webauthn"
 	"github.com/gofrs/uuid"
 	"github.com/pkg/errors"
 	"github.com/supabase/auth/internal/storage"
 	"time"
 )
 
 type Challenge struct {
-	ID         uuid.UUID  `json:"challenge_id" db:"id"`
-	FactorID   uuid.UUID  `json:"factor_id" db:"factor_id"`
-	CreatedAt  time.Time  `json:"created_at" db:"created_at"`
-	VerifiedAt *time.Time `json:"verified_at,omitempty" db:"verified_at"`
-	IPAddress  string     `json:"ip_address" db:"ip_address"`
-	Factor     *Factor    `json:"factor,omitempty" belongs_to:"factor"`
+	ID                uuid.UUID  `json:"challenge_id" db:"id"`
+	FactorID          uuid.UUID  `json:"factor_id" db:"factor_id"`
+	CreatedAt         time.Time  `json:"created_at" db:"created_at"`
+	VerifiedAt        *time.Time `json:"verified_at,omitempty" db:"verified_at"`
+	IPAddress         string     `json:"ip_address" db:"ip_address"`
+	Factor            *Factor    `json:"factor,omitempty" belongs_to:"factor"`
+	WebauthnChallenge *string    `json:"webauthn_challenge,omitempty" db:"webauthn_challenge"`
+	UserVerification  *string    `json:"user_verification,omitempty" db:"user_verification"`
```

Comment:
What is this for?

Reply:
Storing the Challenge and user verification preference. Though, if that is specified by the user then there would be no need for that.

Reply:
Actually, on further thought we will need the Challenge and `UserVerification` on the server

Reply:
As discussed, let's try to store the challenge only